tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs

```diff
@@ -497,6 +488,23 @@ public async Task HandlePlaybackRequest(string recordingId, HttpRequest incoming
 
                 await WriteBodyBytes(bodyData, session.PlaybackResponseTime, outgoingResponse);
             }
+
+            Interlocked.Increment(ref Startup.RequestsPlayedBack);
+
+            var remove = true;
+
+            // If request contains "x-recording-remove: false", then request is not removed from session after playback.
+            // Used by perf tests to play back the same request multiple times.
+            if (incomingRequest.Headers.TryGetValue("x-recording-remove", out var removeHeader))
+            {
+                remove = bool.Parse(removeHeader);
+            }
+
+            // Only remove session once body has been written, to minimize probability client retries but test-proxy has already removed the session
+            if (remove)
+            {
+                session.Session.Lookup(entry, session.CustomMatcher ?? Matcher, session.AdditionalSanitizers.Count > 0 ? Sanitizers.Concat(session.AdditionalSanitizers) : Sanitizers, remove: true);
```

Comment:
Might be able to improve perf, by adding something like a `Remove()` method to `Session`, which safely removes a session (under locks), but doesn't need to re-run the lookup and sanitizer code.